main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -590,6 +586,17 @@ object Type {
     }
 
     override def hashCode(): Int = Objects.hash(tpe1, tpe2)
+
+    /**
+      * Creates a new Type.Apply, reusing this one if they are equivalent.
+      */
+    def renew(newTpe1: Type, newTpe2: Type, newLoc: SourceLocation): Type.Apply = {
```

Comment:
How is it obvious that "renew" and "apply" are related?

Reply:
You mean `Type.Apply`?

This function is only defined on `Type.Apply`, not on `Type`.

Reply:
> You mean `Type.Apply`?
> 
> This function is only defined on `Type.Apply`, not on `Type`.

Smart. I was tired.